### PR TITLE
Test app init

### PR DESCRIPTION
**Pull request by y-lohse**
Review commit by commit, most of the changes is just one big code move.

### CODE REVIEW

File: src/drive/targets/mobile/InitAppMobile.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  it('should not restart the app while its starting', async () => {
+    const app = new InitAppMobile()
+    const startApplicationSpy = jest.spyOn(app, 'startApplication')
+
+    const appStarting = app.initialize()
+    expect(startApplicationSpy).not.toHaveBeenCalled()
+    app.onDeviceReady()
+    expect(startApplicationSpy).not.toHaveBeenCalled()
+    app.onDeviceReady()
+    expect(startApplicationSpy).not.toHaveBeenCalled()
```
Comment (Crash--) [inline src/drive/targets/mobile/InitAppMobile.spec.jsx]: OWI *_*

File: src/drive/mobile/lib/__mocks__/cozy-helper.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const mockClient = {
```
Comment (Crash--) [inline src/drive/mobile/lib/__mocks__/cozy-helper.js]: mockStackClient

File: src/drive/targets/mobile/InitAppMobile.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    app.onDeviceReady()
+    await appStarting
+    expect(startApplicationSpy).toHaveBeenCalled()
+  })
+  it('should inform us when the app is started', async () => {
+    const app = new InitAppMobile()
+    const appStarting = app.initialize()
+    app.onDeviceReady()
+    const result = await appStarting
+    expect(result).toBe(undefined)
```
Comment (Crash--) [inline src/drive/targets/mobile/InitAppMobile.spec.jsx]: should we return true or something ?

File: src/drive/targets/mobile/InitAppMobile.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    render: jest.fn((app, node, afterRender) => {
+      afterRender()
+    })
+  }
+})
+
+describe('App initialize', () => {
+
+  beforeAll(() => {
+    global.__DEVELOPMENT__ = false
```
Comment (cedricmessiant) [inline src/drive/targets/mobile/InitAppMobile.spec.jsx]: reset it to its value `afterAll` ? Or maybe it could go in global jest config?

**Pull request merged**